hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/search/LastNAggregation.java

```diff
@@ -0,0 +1,169 @@
+package ca.uhn.fhir.jpa.dao.search;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.hibernate.search.engine.search.aggregation.AggregationKey;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Builds lastN aggregation, and parse the results
+ */
+public class LastNAggregation {
+	static final String SP_SUBJECT = "sp.subject.reference.value";
+	private static final String SP_CODE_TOKEN_CODE_AND_SYSTEM = "sp.code.token.code-system";
+	private static final String SP_DATE_DT_UPPER = "sp.date.dt.upper";
+	private static final String GROUP_BY_CODE_SYSTEM_SUB_AGGREGATION = "group_by_code_system";
+	private static final String MOST_RECENT_EFFECTIVE_SUB_AGGREGATION = "most_recent_effective";
+
+	private final int myLastNMax;
+	private final boolean myAggregateOnSubject;
+	private final Gson myJsonParser = new Gson();
+
+	public LastNAggregation(int theLastNMax, boolean theAggregateOnSubject) {
+		myLastNMax = theLastNMax;
+		myAggregateOnSubject = theAggregateOnSubject;
+	}
+
+	/**
+	 * Aggregation template json.
+	 * <p>
+	 * https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html
+	 */
+	public JsonObject toAggregation() {
+		JsonObject lastNAggregation = myJsonParser.fromJson(
+			"{\n" +
+				"   \"terms\":{\n" +
+				"      \"field\":\"" + SP_CODE_TOKEN_CODE_AND_SYSTEM + "\",\n" +
+				"      \"size\":100,\n" +
+				"      \"min_doc_count\":1\n" +
+				"   },\n" +
+				"   \"aggs\":{\n" +
+				"      \"" + MOST_RECENT_EFFECTIVE_SUB_AGGREGATION + "\":{\n" +
+				"         \"top_hits\":{\n" +
+				"            \"size\":" + myLastNMax + ",\n" +
```

Comment:
`myLastNMax` is an int.  Do you mean ?
```suggestion
				"            \"size\": + myLastNMax + ,\n" +
```